.github/workflows/build_infra_images_cache.yml

```diff
@@ -27,6 +27,7 @@ on:
     - 'branch-*'
     paths:
     - 'dev/infra/Dockerfile'
+    - 'dev/spark-test-image/docs/Dockerfile'
```

Comment:
IIRC, last time, this line seems to fail at `branch-3.5` run because this file doesn't exist in `branch-3.5`.

Did you get a chance to test this with `branch-3.5` since last reverting, @zhengruifeng ?

Reply:
Not yet, because I am not sure how to test again `branch-3.5` in my branches.

After revert, I made such changes in this PR:
1, keep the location of previous image `dev/infra/Dockerfile`, it will be used for both 3.5, and master (except docs);
2, add file existence check before each step needs the new docs image;
3, additional url like for docs jobs, it links to previous image for `branch-3.5`, and the new image for `master`.

Reply:
BTW I guess this line should be fine, because the path in `paths` here does not need to be existent.

https://github.com/apache/spark/pull/48533/files fixed `ERROR: failed to solve: failed to read dockerfile: open Dockerfile: no such file or directory` in https://github.com/apache/spark/actions/runs/11387123331/job/31682246504

It failed in `Build and Push`, not the `paths`